ui/perfherder/graphs/TableView.jsx

```diff
@@ -1,21 +1,77 @@
 import React from 'react';
 import ReactTable from 'react-table';
+import { Badge } from 'reactstrap';
 import moment from 'moment';
 import { groupBy, forIn } from 'lodash';
 import numeral from 'numeral';
 
 import RepositoryModel from '../../models/repository';
 import { getJobsUrl, createQueryParams } from '../../helpers/url';
-import { displayNumber } from '../helpers';
+import { getFrameworkName, displayNumber } from '../helpers';
 
 const TableView = ({
   testData,
   highlightAlerts,
   highlightedRevisions,
   projects,
+  frameworks,
 }) => {
-  const allDataPoints = [];
-  const tableDataPoints = [];
+  let tableDataPoints = [];
+  let columns = [];
+
+  const getRevisionInfo = (dataIndex, dataPoint, item) => {
+    const prevFlotDataPointIndex = dataIndex - 1;
+
+    // calculate delta values
+    const valueY = dataPoint.y;
+    const v0 =
+      prevFlotDataPointIndex !== -1
+        ? item.data[prevFlotDataPointIndex].y
+        : valueY;
+    const deltaValue = valueY - v0;
+    const deltaPercent = valueY / v0 - 1;
+
+    // get push link
+    let prevRevision;
+    let pushUrl;
+    if (prevFlotDataPointIndex !== -1) {
+      prevRevision = item.data[prevFlotDataPointIndex].revision;
```

Comment:
Some of this logic from the tooltip can be put into a helper function to be used by both components. That way if it needs to be changed in the future, you only have to change it in that helper function.

Reply:
@yogmel, @sarah-clements  just be aware that there's an [open PR 6066](https://github.com/mozilla/treeherder/pull/6066/files) that is fixing the [empty pushlogs](https://hg.mozilla.org/integration/autoland/pushloghtml?fromchange=458b5c5e018362ad113be1baf2feb4bdc2913666&tochange=458b5c5e018362ad113be1baf2feb4bdc2913666) from [graph tooltip](https://treeherder.mozilla.org/perf.html#/graphs?highlightAlerts=1&selected=1926849,1057676531&series=autoland,1926849,1,1&timerange=1209600&zoom=1582199603282,1582307149191,312.513848695788,364.11166716660455) that appears from the second data-point (first retrigger) of a revision. This is because `prevRevision` is taking the `revision` hash from `prevFlotDataPointIndex`, and if the revision is the same, the pushlog link generated is not correct.

Moving this to a helper function might make fixing this bug harder if it doesn't pass `dataPointDetails` and `testDetails` as params. I don't mind either if you include the fix into this PR. Whatever works best for you. Thanks